src/stateProviders/provideFenceConnections.js

```diff
@@ -0,0 +1,92 @@
+import { provideState, update } from 'freactal';
+
+import { getFenceUser } from 'services/fence';
+import { getUserStudyPermission } from 'services/fileAccessControl';
+import { FENCES } from 'common/constants';
+import { omit, flatMap, isEmpty } from 'lodash';
+
+export default provideState({
+  initialState: props => ({
+    fenceConnectionsInitialized: false,
+    fenceConnections: {},
+
+    fenceStudiesInitialized: false,
+    fenceStudies: {},
+  }),
+  computed: {
+    fenceAuthStudies: ({ fenceStudies }) =>
+      !isEmpty(fenceStudies) &&
+      flatMap(Object.values(fenceStudies), studies => studies.authorizedStudies),
+    fenceNonAuthStudies: ({ fenceStudies }) =>
+      !isEmpty(fenceStudies) &&
+      flatMap(Object.values(fenceStudies), studies => studies.unauthorizedStudies),
+  },
+  effects: {
+    setFenceConnectionsInitialized: update(state => ({
+      fenceConnectionsInitialized: true,
+    })),
+
+    clearFenceConnections: update((state, fence) => ({
+      fenceConnections: {},
+      fenceStudies: {},
+    })),
+    addFenceConnection: update((state, { fence, details }) => {
+      return {
+        fenceConnections: { ...state.fenceConnections, [fence]: details },
+      };
+    }),
+    removeFenceConnection: update((state, fence) => ({
+      fenceConnections: omit(state.fenceConnections, fence),
+      fenceStudies: omit(state.fenceStudies, fence),
+      fenceStudiesInitialized: false,
+    })),
+    fetchFenceConnections: (effects, { api }) => {
+      const fenceConnectionsFetchArray = FENCES.map(fence =>
+        getFenceUser(api, fence)
+          .then(details => {
+            effects.addFenceConnection({ fence, details });
+          })
+          .catch(err => console.log(`Error fetching fence connection for '${fence}': ${err}`)),
+      );
+
+      return Promise.all(fenceConnectionsFetchArray).then(() => {
+        effects.setFenceConnectionsInitialized();
+        effects.setFenceStudiesInitialized(false);
+      });
+    },
```

Comment:
would it make sense to handle the fetching in the [initialize](https://github.com/FormidableLabs/freactal#initialize) effect rather than letting a wrapped component trigger this? Same for `fetchFenceStudies`

Reply:
It does make sense, but a couple tricks - it is dependant on the user being logged in so step one is move the provider to a page component that only loads once the user is logged in, and then we need a way to provide the initialized API object that has the JWT auth. If theres a clean way to satisfy those dependencies than I would prefer to run this as initialized, but we'll also need to trigger it on log in as well.

Note that we will still need to trigger the studies load whenever they add a new connection.